### ISSUE TITLE
<locale>: std::ctype<char>::blank return true for all space-characters

### ISSUE THREAD

**Issue by fsb4000** [microsoft/STL#1121]
**Describe the bug**
blank returns true for '\r', '\n' ... it just works as an alias for std::ctype<char>::space

From https://en.cppreference.com/w/cpp/string/byte/isblank

> In the default C locale, only space (0x20) and horizontal tab (0x09) are classified as blank characters. 

**Command-line test case**
```
C:\Temp>type main.cpp
#include <iostream> 
#include <locale> 
#include <cctype>

using namespace std;

int main()
{ 
	cout << isblank('\n') << '\n';
	cout << use_facet<ctype<char>>(locale("C")).is(ctype<char>::blank,
		static_cast<unsigned char>('\n')) << '\n';
	cout << use_facet<ctype<char>>(locale::classic()).is(ctype<char>::blank,
		static_cast<unsigned char>('\n')) << '\n';
}

C:\Temp>cl /EHsc /W4 /WX main.cpp
cl /EHsc /W4 /WX main.cpp
Оптимизирующий компилятор Microsoft (R) C/C++ версии 19.27.29109 для x64
(C) Корпорация Майкрософт (Microsoft Corporation).  Все права защищены.

main.cpp
Microsoft (R) Incremental Linker Version 14.27.29109.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:main.exe
main.obj

C:\Temp>main.exe
0
1
1
```

**Expected behavior**
Should return 0 0 0

**STL version**
    ```
    Microsoft Visual Studio Community 2019 Preview
    Version 16.7.0 Preview 6.0
    ```
**Additional context**
[VSO-429255](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/429255) | [DevCom-215436](https://developercommunity.visualstudio.com/content/problem/215436/meow.html) [DevCom-246252](https://developercommunity.visualstudio.com/content/problem/246252/meow.html) [DevCom-51443](https://developercommunity.visualstudio.com/content/problem/51443/meow.html)

**Comment by Agrael1** [comment]
I've investigated the issue, it'll require rework on ucrt side of things, also this may break binary compatibility since the comparison flags may change.

**Comment by Arzaghi** [comment]
I've investigated this issue and spent some time on it. I agree with @agrael1. I think it's impossible to fix it without touching the following `enum`. I guess it breaks the ABI (?). 

https://github.com/microsoft/STL/blob/03ab954ea23fec65fb86648aae016e4aa12a3e6d/stl/inc/xlocale#L2359-L2380


If changing the above `enum` does break the binary compatibility then I think it should be marked as `vNext` and if it doesn't break ABI, please let me know. I can continue my draft PR.

**Comment by Agrael1** [comment]
Yep, I asked for vNext, but somehow flag wasn't set, issue was passed to ucrt team

**Comment by CraigHutchinson** [comment]
I just experienced this issue `std::isblank('\n', std::locale::classic())` == `true` which is very unexepected!

**Comment by CraigHutchinson** [comment]
> I've investigated this issue and spent some time on it. I agree with @Agrael1. I think it's impossible to fix it without touching the following `enum`. I guess it breaks the ABI (?).
> 
> https://github.com/microsoft/STL/blob/03ab954ea23fec65fb86648aae016e4aa12a3e6d/stl/inc/xlocale#L2359-L2380
> 
> If changing the above `enum` does break the binary compatibility then I think it should be marked as `vNext` and if it doesn't break ABI, please let me know. I can continue my draft PR.

As `space` and `blank` are the same value changing that enum isn't an ABI break from my understanding? i.e. Existing ABI calls are to `_CN | _SP` so changing the `blank` in this enum won't affect existing afaik. Not sure why this remains unfixed!

**Comment by StephanTLavavej** [comment]
We are more daring about behavioral changes these days - I suspect that fixing this can be done in an ABI-compatible way. Because it touches locales it is definitely one of the scarier outstanding issues though.
